File: tests/nodejs/cases/case_040_normalize_ledger.js
/**
 * Task case 040: normalize_ledger
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case040NormalizeLedger(input) {
  // Task: normalize ledger records and prepare transport-ready payload.
  const taskName = 'normalize_ledger';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'ledger',
    stage: 'normalize',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // SAFE: payload remains local; only primitive checksum string is returned.
  return payload.checksum;
}

module.exports = {
  case040NormalizeLedger,
};

File: tests/python/cases/case_040_normalize_ledger.py
"""
Task case 040: normalize ledger payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_040_normalize_ledger(input_data):
    # Task: normalize ledger records and prepare transport-ready payload.
    task_name = 'normalize_ledger'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'ledger',
        'stage': 'normalize',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # SAFE: payload remains local; only primitive checksum string is returned.
    return payload['checksum']
